edb/edgeql/compiler/setgen.py

```diff
@@ -264,6 +264,15 @@ def compile_path(expr: qlast.Path, *, ctx: context.ContextLevel) -> irast.Set:
             if ptr_expr.type == 'property':
                 # Link property reference; the source is the
                 # link immediately preceding this step in the path.
+                if path_tip.rptr is None:
+                    source = get_set_type(path_tip, ctx=ctx)
+                    s_dn = source.get_displayname(ctx.env.schema)
+                    raise errors.InvalidReferenceError(
```

Comment:
I'd use "unexpected reference to link property {ptr_name!r} outside of a path expression", and I think it should be an `EdgeQLSyntaxError` (we normally use `InvalidReferenceError` to indicate a syntactically-correct reference to a non-existent object).

Reply:
I've changed error and message.

> we normally use `InvalidReferenceError` to indicate a syntactically-correct reference to a non-existent object

Thanks for the explanation. I raised an `InvalidReferenceError` following the message in the issue, but the syntax error here really looks more appropriate, given what is written in the issue:
> The only valid form of link property reference is .link@prop.